ProcessMaker/processmaker
# Refactor Template Media Handling and Resolve Thumbnail Deletion Error

This PR refactors how template media is uploaded, stored, and retrieved, while also resolving an error occurring during the deletion of an image from the MultiThumbnail Image preview component. 


## Solution

- Removal of duplicate code responsible for retrieving template media, replaced with a unified approach to handle both PLG screen template media and user-created screen template media.
- Elimination of unused API routes and API calls related to deleting template media.
- Implementation of procedures for syncing template media when template configurations are updated.

## How to Test

1. Create a Screen Template and add/remove thumbnails.
2. Ensure no errors occur when removing thumbnails.
3. Save the template.
4. Navigate to the screen template and select 'Configure'.
5. Verify that the template media displays correctly.
6. Add/remove media and ensure no errors occur.
7. Select 'Save' and ensure no errors are encountered.
8. Navigate to 'Designer > Screens'.
9. Select '+ Screen' and navigate to the screen template.
10. Verify that the appropriate thumbnail is displayed.
11. Select 'Preview Template' and ensure that the correct thumbnails are previewed.

## Related Tickets & Packages
- [FOUR-14664](https://processmaker.atlassian.net/browse/FOUR-14664)

ci:next
ci:deploy

## Code Review Checklist


[FOUR-14664]: https://processmaker.atlassian.net/browse/FOUR-14664?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
